File: .codex/hooks/hook_response.py
#!/usr/bin/env python3
"""Shared Codex/Claude hook response envelopes."""

from __future__ import annotations


def deny(
    reason: str,
    *,
    event_name: str = "PreToolUse",
) -> dict[str, object]:
    """Return a PreToolUse-style denial with a human-readable reason."""
    return {
        "hookSpecificOutput": {
            "hookEventName": event_name,
            "permissionDecision": "deny",
            "permissionDecisionReason": reason,
        }
    }


def additional_context(
    context: str,
    *,
    event_name: str,
) -> dict[str, object]:
    """Return an additionalContext payload for the given hook event name."""
    return {
        "hookSpecificOutput": {
            "hookEventName": event_name,
            "additionalContext": context,
        }
    }

File: .codex/hooks/lib/hook_response.ts
export const VERSION = "1.0.0";

export function deny(
  reason: string,
  eventName = "PreToolUse",
): Record<string, unknown> {
  return {
    hookSpecificOutput: {
      hookEventName: eventName,
      permissionDecision: "deny",
      permissionDecisionReason: reason,
    },
  };
}

export function additionalContext(
  context: string,
  eventName: string,
): Record<string, unknown> {
  return {
    hookSpecificOutput: {
      hookEventName: eventName,
      additionalContext: context,
    },
  };
}
